app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -523,6 +525,17 @@ public void updateFailingUrl(String url, boolean updateFromError) {
                     this.failingUrl = url;
                 }
             }
+
+            @Override
+            public void onReceivedTitle(WebView view, String title) {
+                if(!mostOldCallbacksHaveFinished) {
+                    return;
+                }
+
+                if (!UrlUtils.isInternalErrorURL(view.getUrl())) {
```

Comment:
Let's just move this check into updateUrl() to avoid duplicating these and simply call onUrlChanged() without the check here?